lib/AcqList/hooks/useLocationFilters.js

```diff
@@ -80,6 +82,27 @@ const useLocationFilters = (location, history, resetData) => {
     [history, resetFilters],
   );
 
+  const clearLocationFilters = useCallback(
+    () => {
+      /*
+        there's a lot of state setters called synchronously both in this hook and it's consumers
+        so we need to be careful about not using old state data
+      */
+      setFilters((currentFilters) => {
+        const newFilters = pick(currentFilters, ['qindex', 'query']);
```

Comment:
It is necessary to reset all states in `filters`, we just need to keep the input state (`searchQuery`) and we also can keep `qindex` (`searchIndex`), but it is not required it will be changed anyway.

This might cause this [unexpected records call](https://github.com/folio-org/stripes-acq-components/assets/77053927/8471b198-e66b-4c5f-92f3-78c7c16482ff).

I suggest doing it like this:
```
setFilters({});
history.push({
  pathname: '',
  search: '',
});
```

Reply:
But `history.push` will reset both `qindex` and `query`
We just need to remove filter state from `filters` object so when we switch between browse options - not supported filters are not applied
`qindex` and `query` parameters will be passed to the `useInventoryBrowse` hook to be used to build request parameters

Keep in mind that when switching between browse options we need to:
1. keep query in the search box
2. keep current results when possible (for example when switching between Call numbers and Contributors they have different filters so we need to clear filters to we'll get results from last Contributors browse)